### PR TITLE
soc: arm: Add support for STM32H730xxQ

### PR DESCRIPTION
**Pull request by xingrz**
The STM32H730 series has a variant built with SMPS. It uses [`stm32h730xxq.h`](https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32h7xx/soc/stm32h730xxq.h) header file instead of `stm32h730xx.h`, which has the `SMPS` macro defined.

This PR adds the `SOC_STM32H730XXQ` configuration option to allow the build system include the proper header file. With this change, boards can enable `CONFIG_POWER_SUPPLY_DIRECT_SMPS` to [set up](https://github.com/zephyrproject-rtos/zephyr/blob/419dda3ed72d968e64d474994f13b7ecb623f18b/soc/arm/st_stm32/stm32h7/soc_m7.c#L96) the power supply for the CPU.

### CODE REVIEW

**Comment by erwango** [comment]
Thanks for this addition.
Please fix compliance check issue

**Comment by xingrz** [comment]
@erwango Thanks for your advice. I just updated the PR.

File: soc/arm/st_stm32/stm32h7/Kconfig.defconfig.stm32h730xxq [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# ST STM32H730XXQ MCU configuration options
+
+# Copyright (c) 2023 Chen Xingyu <hi@xingrz.me>
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_STM32H730XXQ
```
Comment (FRASTM) [inline soc/arm/st_stm32/stm32h7/Kconfig.defconfig.stm32h730xxq:6]: Any comment could help to explain what a new serie is added here which is exactly the same as the existing `STM32H730XX`

**Review by FRASTM — COMMENTED** [review]
LGTM

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**